jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/JPAClassFileTransformerProcessor.java

```diff
@@ -40,14 +40,13 @@ private void setClassLoaderTransformer(DeploymentUnit deploymentUnit) {
         // during the call to CreateContainerEntityManagerFactory.
 
         DelegatingClassTransformer transformer = deploymentUnit.getAttachment(DelegatingClassTransformer.ATTACHMENT_KEY);
-        boolean appContainsPersistenceProviderJars = false;  // remove when we revert WFLY-10520
         if ( transformer != null) {
 
             for (ResourceRoot resourceRoot : DeploymentUtils.allResourceRoots(deploymentUnit)) {
                 PersistenceUnitMetadataHolder holder = resourceRoot.getAttachment(PersistenceUnitMetadataHolder.PERSISTENCE_UNITS);
                 if (holder != null) {
                     for (PersistenceUnitMetadata pu : holder.getPersistenceUnits()) {
-                        if (Configuration.needClassFileTransformer(pu)) {
+                        if (Configuration.needClassFileTransformer(pu) && pu.needsJPADelegatingClassFileTransformer()) {
```

Comment:
Minor nit, I'd do the `pu.needsJPADelegatingClassFileTransformer()` because it's a faster check. If it returns `false`, then the next check won't be done.

Reply:
Good idea, I'll rename the `needsJPADelegatingClassFileTransformer` to better this change exactly one thread will  find that Configuration.needClassFileTransformer(pu) returns true or false.

Reply:
Well not that exactly but will make a change.